streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -133,7 +139,12 @@ public void flush(final String namespace) {
         if (cache == null) {
             return;
         }
-        cache.flush();
+
+        synchronized (cache) {
+            final long oldSize = cache.sizeInBytes();
+            cache.flush();
```

Comment:
ditto here, `#flush` could return the number of bytes flushed/evicted -- although I guess in this case the overhead saved is less because we already hold the lock on the `cache` itself

Still, it might encourage more sensible bookkeeping of the cache sizes to always return the number of bytes that changed for a given call that was return type `void` anyways